src/components/EnterpriseApp/index.jsx

```diff
@@ -125,6 +126,10 @@ class EnterpriseApp extends React.Component {
       return <EnterpriseAppSkeleton />;
     }
 
+    if (!loading && enterpriseId === null) {
+      return <NotFoundPage />;
```

Comment:
It's probably worth adding a test case in `EnterpriseApp.test.jsx` for this scenario of rendering the `NotFoundPage`.